### PR TITLE
refactor: reorganize addBarmanEndpointCAToJob function

### LINKED ISSUE #361: refactor addBarmanEndpointCAToJob function
Right now the function:

https://github.com/cloudnative-pg/cloudnative-pg/blob/1a322fa7a703c232b401b4fa7e50db676c3a1d9f/pkg/specs/jobs.go#L135

Is extremely hard to maintain given to the approach used. Adding new providers would be extremely error prone

We should refactor and make it more maintainable.

### PR DESCRIPTION
**Pull request by armru**
This patch reorganizes in a more structured way the fields related to the barman credentials and how they are used to build the environment variables for the instance.

Closes #361 

Signed-off-by: Armando Ruocco <armando.ruocco@enterprisedb.com>

### CODE REVIEW

File: pkg/specs/pods.go
Comment (armru) [inline pkg/specs/pods.go:351]: This is intentionally exported

**Comment by jlong49** [comment]
> @jlong49, here's the link to the E2E on CNPG workflow run: https://github.com/cloudnative-pg/cloudnative-pg/actions/runs/2699817277

One test failed in this run but it was just a timeout and I re-ran the test locally and it passed

**Review by jlong49 — APPROVED** [review]

**Comment by leonardoce** [comment]
I added a review commit that is really a refactoring to avoid something like #453 to happen again.

From the serialization PoV nothing changes but people using this codebase as a dependency will need to change a bit how they are creating clusters. We may decide to keep my review commit or remove it, anyway I'd like to have your feedback guys!

**Review by mnencia — APPROVED** [review]

**Comment by mnencia** [comment]
The PR is OK for me. I added the "do not merge" label because I want @armru to review it before the final merge. @armru, once satisfied, please remove the label and merge it.

**Pull request merged**
